contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+            );
+
+        require(_wethAmount == 0, "Non zero wethAmount");
+        underlyingAssets = _oethbAmount;
+        emit UnderlyingAssetsUpdated(underlyingAssets);
+    }
+
+    /**
+     * @notice Deposit WETH to the strategy contract. This function does not add liquidity to the
+     *         underlying Aerodrome pool.
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (_wethBalance > 0) {
+            _deposit(WETH, _wethBalance);
+        }
+    }
+
+    /**
+     * @notice Withdraw an `amount` of assets from the platform and
+     *         send to the `_recipient`.
+     * @param _recipient  Address to which the asset should be sent
+     * @param _asset      WETH address
+     * @param _amount     Amount of WETH to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_asset == WETH, "Unsupported asset");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (_wethBalance < _amount) {
+            uint256 _additionalWethRequired = _amount - _wethBalance;
+            (uint256 _wethInThePool, ) = getPositionPrincipal();
+
+            if (_wethInThePool < _additionalWethRequired) {
+                revert NotEnoughWethLiquidity(
+                    _wethInThePool,
+                    _additionalWethRequired
+                );
+            }
+
+            uint256 shareOfWethToRemove = Math.min(
+                _additionalWethRequired.divPrecisely(_wethInThePool) + 1,
+                1e18
+            );
+            _removeLiquidity(shareOfWethToRemove);
+        }
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        _withdraw(_recipient, _amount);
+    }
+
+    /**
+     * @notice Withdraw WETH and sends it to the Vault.
+     */
+    function withdrawAll() external override onlyVault nonReentrant {
+        if (tokenId > 0) {
+            _removeLiquidity(1e18);
+        }
+
+        uint256 _balance = IERC20(WETH).balanceOf(address(this));
+        if (_balance > 0) {
+            _withdraw(vaultAddress, _balance);
+        }
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+    }
+
+    function _withdraw(address _recipient, uint256 _amount) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        IERC20(WETH).safeTransfer(_recipient, _amount);
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+    /**
+     * @dev Collect the AERO token from the gauge
+     */
+    function _collectRewardTokens() internal override {
+        if (tokenId > 0) {
+            clGauge.getReward(tokenId);
```

Comment:
nit: `CLGauge` has a `notifyRewardAmount` to trigger the reward emissions, might be worth calling it here before `getReward`. However, it's gonna get collected during the next collect anyway, so probably doesn't matter much

Reply:
I am not 100% on this one, though it seems to me that calling this function is the responsibility of the voters when they delegate Aero emissions to a certain pool on the gauge.
If the strategy contract makes this call it reverts [on this line](https://github.com/aerodrome-finance/slipstream/blob/main/contracts/gauge/CLGauge.sol#L275) with the "NV" which seems to indicate `not a voter` error

Reply:
Ah, my bad then